packages/expo-sqlite/ios/SQLiteModule.swift

```diff
@@ -0,0 +1,209 @@
+    }
+
+    if db == nil {
+      cachedDatabases.removeValue(forKey: dbName)
+      if sqlite3_open(path.absoluteString, &db) != SQLITE_OK {
+        return nil
+      }
+
+      cachedDatabases[dbName] = db
+    }
+    return db
+  }
+
+  private func executeSql(sql: String, with args: [Any], for db: OpaquePointer, readOnly: Bool) -> [Any?] {
+    var resultRows = [Any]()
+    var statement: OpaquePointer?
+    var rowsAffected: Int32 = 0
+    var insertId: Int64 = 0
+    var error: String?
+
+    if sqlite3_prepare_v2(db, sql, -1, &statement, nil) != SQLITE_OK {
+      return [converSqlLiteErrorToString(db: db)]
+    }
+
+    let queryIsReadOnly = sqlite3_stmt_readonly(statement) > 0
+
+    if readOnly && !queryIsReadOnly {
+      return ["could not prepare \(sql)"]
+    }
+
+    for (index, arg) in args.enumerated() {
+      guard let obj = arg as? NSObject else { continue }
+      bindStatement(statement: statement, with: obj, at: Int32(index + 1))
+    }
+
+    var columnCount: Int32 = 0
+    var columnNames = [String]()
+    var columnType: Int32
+    var fetchedColumns = false
+    var value: Any?
+    var hasMore = true
+
+    while hasMore {
+      let result = sqlite3_step(statement)
+
+      switch result {
+      case SQLITE_ROW:
+        if !fetchedColumns {
+          columnCount = sqlite3_column_count(statement)
+
+          for i in 0..<Int(columnCount) {
+            let columnName = NSString(format: "%s", sqlite3_column_name(statement, Int32(i))) as String
+            columnNames.append(columnName)
+          }
+          fetchedColumns = true
+        }
+
+        var entry = [Any]()
+
+        for i in 0..<Int(columnCount) {
+          columnType = sqlite3_column_type(statement, Int32(i))
+          value = getSqlValue(for: columnType, with: statement, index: Int32(i))
+          entry.append(value)
+        }
+
+        resultRows.append(entry)
+      case SQLITE_DONE:
+        hasMore = false
+      default:
+        error = converSqlLiteErrorToString(db: db)
+        hasMore = false
+      }
+    }
+
+    if !queryIsReadOnly {
+      rowsAffected = sqlite3_changes(db)
+      if rowsAffected > 0 {
+        insertId = sqlite3_last_insert_rowid(db)
+      }
+    }
+
+    sqlite3_finalize(statement)
+
+    if error != nil {
+      return [error]
+    }
+
+    return [nil, insertId, rowsAffected, columnNames, resultRows]
+  }
+
+  private func bindStatement(statement: OpaquePointer?, with arg: NSObject, at index: Int32) {
+    if arg == NSNull() {
+      sqlite3_bind_null(statement, index)
+    } else if arg is Double {
+      sqlite3_bind_double(statement, index, arg as? Double ?? 0.0)
+    } else {
+      var stringArg: NSString
+
+      if arg is NSString {
+        stringArg = NSString(format: "%@", arg)
+      } else {
+        stringArg = arg.description as NSString
+      }
+
+      let data = stringArg.data(using: NSUTF8StringEncoding)
+      sqlite3_bind_text(statement, index, stringArg.utf8String, -1, nil)
```

Comment:
is it intentional to pass `-1, nil`?

reference for original code:
https://github.com/expo/expo/blob/6afa4eb4d32197682433fab743949aa872d431d2/packages/expo-sqlite/ios/EXSQLite/EXSQLite.m#L252

Reply:
Yes, this was intentional. A negative value means bytes up to the first terminating 0. This seems to be the convention when using this API in Swift. I could do this if you prefer.
```swift
sqlite3_bind_text(statement, index, stringArg.utf8String, Int32(data?.count ?? 0), nil)
```

SQLITE_TRANSIENT is not imported in swift so it is passed as nil. I could also define it myself and use it that way
```swift
let SQLITE_TRANSIENT = unsafeBitCast(OpaquePointer(bitPattern: -1), to: sqlite3_destructor_type.self)
sqlite3_bind_text(statement, index, stringArg.utf8String, Int32(data?.count ?? 0), SQLITE_TRANSIENT)
```